NineChronicles.Headless.Executable/Commands/ReplayCommand.cs

```diff
@@ -240,9 +184,332 @@ public void Help([FromService] ICoconaHelpMessageBuilder helpMessageBuilder)
+                        {
+                            blockChain.ExecuteActions(
+                                block,
+                                StateCompleterSet<NCAction>.Reject);
+
+                            if (verbose)
+                            {
+                                msg = "(o)";
+                                _console.Out.Write(msg);
+                                outputSw?.Write(msg);
+                            }
+                        }
+                        catch (InvalidBlockStateRootHashException)
+                        {
+                            if (!verbose)
+                            {
+                                throw;
+                            }
+
+                            msg = "(x)";
+                            _console.Out.WriteLine(msg);
+                            outputSw?.WriteLine(msg);
+
+                            var actionEvaluator = GetActionEvaluator(
+                                blockChain,
+                                stateStore,
+                                blockChain.Genesis.Hash);
+                            var actionEvaluations = actionEvaluator.Evaluate(
+                                block,
+                                StateCompleterSet<NCAction>.Reject);
+                            for (var k = 0; k < actionEvaluations.Count; k++)
+                            {
+                                var actionEvaluation = actionEvaluations[k];
+                                var prefix = $"--- action evaluation #{k + 1}" +
+                                             $" tx-id({actionEvaluation.InputContext.TxId})";
+                                msg = actionEvaluation.Exception is null
+                                    ? $"{prefix} exception: null"
+                                    : $"{prefix} exception: {actionEvaluation.Exception}";
+                                _console.Out.WriteLine(msg);
+                                outputSw?.WriteLine(msg);
+                                LoggingAboutIncompleteBlockStatesException(
+                                    actionEvaluation.Exception,
+                                    currentBlockIndex - 1,
+                                    storePath,
+                                    outputSw,
+                                    $"{prefix} tip: ");
+                            }
+
+                            msg = $"- block #{block.Index} evaluating failed with ";
+                            _console.Out.Write(msg);
+                            outputSw?.Write(msg);
+
+                            throw;
+                        }
+                        catch (Exception)
+                        {
+                            if (!verbose)
+                            {
+                                throw;
+                            }
+
+                            _console.Out.Write("\n");
+                            outputSw?.Write("\n");
```

Comment:
Don't we have to write down what kind of exception has been occurred?

Reply:
These lines was fixed!